bazel/repository_locations.bzl

```diff
@@ -287,6 +287,28 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         release_date = "2020-11-23",
         cpe = "cpe:2.3:a:nghttp2:nghttp2:*",
     ),
+    com_github_nilslice_protolock = dict(
+        project_name = "protolock",
+        project_desc = "Track your .proto files and prevent changes to messages and services which impact API compatibility.",
+        project_url = "https://github.com/nilslice/protolock/",
+        version = "0.15.2",
+        sha256 = "26d0b491471866c4959a75a1418a577eab432782b9923fdf022535bb7c322374",
+        urls = ["https://github.com/nilslice/protolock/archive/refs/tags/v{version}.tar.gz"],
+        release_date = "2021-02-18",
+        use_category = ["api"],
+        strip_prefix = "protolock-{version}",
+    ),
```

Comment:
scorecard --repo=https://github.com/nilslice/protolock
```
RESULTS
-------
Repo: github.com/nilslice/protolock
|--------|------------|-----------------------------|
| STATUS | CONFIDENCE |            NAME             |
|--------|------------|-----------------------------|
| Pass   |         10 | Binary-Artifacts            |
|--------|------------|-----------------------------|
| Pass   |         10 | Code-Review                 |
|--------|------------|-----------------------------|
| Pass   |         10 | Contributors                |
|--------|------------|-----------------------------|
| Pass   |          8 | Pull-Requests               |
|--------|------------|-----------------------------|
| Pass   |         10 | Token-Permissions           |
|--------|------------|-----------------------------|
| Pass   |         10 | Vulnerabilities             |
|--------|------------|-----------------------------|
| Fail   |         10 | Active                      |
|--------|------------|-----------------------------|
| Fail   |          3 | Automatic-Dependency-Update |
|--------|------------|-----------------------------|
| Fail   |          0 | Branch-Protection           |
|--------|------------|-----------------------------|
| Fail   |          5 | CI-Tests                    |
|--------|------------|-----------------------------|
| Fail   |         10 | CII-Best-Practices          |
|--------|------------|-----------------------------|
| Fail   |         10 | Frozen-Deps                 |
|--------|------------|-----------------------------|
| Fail   |         10 | Fuzzing                     |
|--------|------------|-----------------------------|
| Fail   |          0 | Packaging                   |
|--------|------------|-----------------------------|
| Fail   |         10 | SAST                        |
|--------|------------|-----------------------------|
| Fail   |         10 | Security-Policy             |
|--------|------------|-----------------------------|
| Fail   |         10 | Signed-Releases             |
|--------|------------|-----------------------------|
| Fail   |         10 | Signed-Tags                 |
|--------|------------|-----------------------------|

Reply:
Since this is really primarily a CI tool I'm willing to give a pass on most of these I think. @yaseenalk what do you think of Protolock maintenance story? I'm not seeing a lot of activity in past year+ and CI is broken. Would we consider forking if this isn't actively maintained?

Reply:
@YaseenAlk also looked at [buf](https://github.com/bufbuild/buf) which is more actively maintained.
The 2 downsides, IIRC, are that buf also attempts to address more protobuf related operations, and it might require more work to add/modify rules (compared to protolock's plugins).
One thing to note is that [buf's breaking changes rules](https://docs.buf.build/breaking-rules/) seem to be much more comprehensive than protolock's.

Reply:
Yeah, `buf` looks a lot more active / maintained. How much harder is it in reality to do stuff in `buf`?

Reply:
I can get direct help from the buf team if needed. Do you want to talk to them about our needs? I think they would be excited to work with us.

Reply:
Out-of-the-box, buf seems to be targeting many of the issues we would like to guard against.
I guess the open question is how hard will it be to add PGV rules to buf. @YaseenAlk, we can discuss this tomorrow.

Reply:
We actually have a migration doc concerning protolock https://docs.buf.build/migration-protolock including a pros/cons list. You likely want to use buf. Happy to chat!

Reply:
Follow up PR that uses buf instead: #17515